### ISSUE TITLE
SSH auth support with Docker driver

### ISSUE THREAD

**Issue by zstyblik** [jenkinsci/nomad-plugin#21]
Hello,

my apologies, but it's unclear to me whether SSH authentication is or isn't supported for Jenkins slaves when Docker driver is used.

Thank you.

**Comment by zstyblik** [comment]
The answer is no, SSH authentication is not supported at the moment. Now that some time has passed, I see it as a nice to have feature which has its pros, easier firewall management, but probably isn't essential. Also, it's possible it would complicate code logic a lot, but I'm unable to say due to inexperience with Java, nomad and Jenkins plugin programming.

**Comment by jovandeginste** [comment]
@zstyblik What is your use case for SSH authentication?

**Comment by zstyblik** [comment]
Actually, since this plugin doesn't support SSH auth, it's none. However, I had to open up firewall and unrestricted access to Jenkins from all nomad hosts. I don't use auth in Jenkins itself. If I did, it might have been a bit different, although eg. `/computer` isn't amongst URIs which shouldn't require auth.
SSH auth simply has its advantages. One of them is that you can have your Jenkins behind the firewall and allow communication from Jenkins towards nomad, or any host in this matter, but this isn't required the other way around.

**Comment by zstyblik** [comment]
You know, maybe I have use-case after all. jNLP is proving to be really troublesome kind. First, there was issue with flow. Ok, that's bug in some kind of application.

Now, I'm getting `org.jenkinsci.remoting.protocol.NetworkLayer.onRecvClosed` and I'm simply unable to run one Android build for no obvious reason. It can be the same issue as with flow, but I honestly doubt it. With another batch job, jenkins slave simply times-out regardless of pinger. I admit, this has been caused by heavy network maintenance, but still ... does pinger thread work after all?

After reading quite a lot stack-overflow posts and jenkins bugs my conclusion is:

* no easy way to debug
* in most cases, there is no solution
* people, core people, don't seem to care. https://github.com/jenkinsci/jenkins/pull/2601#issuecomment-291010115 has nice touch to it.

In other words, it could be feature worth the shot ... at some of time.
